File: objectorientedprogramming/oops/polymorphism/method_overloading.py
# method overloading
# same method different number of arguments
# method overloading can be existed in single class

class Maths:
    def add(self):
        print("inside no arg")

    def add(self, num1):
        print("inside one arg")

    def add(self, num1, num2):
        print("inside two arg")


m = Maths
# m.add(10)
m.add(10, 20)

File: javascript/objectoriented/polymorphism/methodoverloading.js
//method overloading(same method name and different num of argmnts)
//many forms

class Calc{
    add(){
        console.log("inside no arg method");
    }
    add(num){
        console.log("inside one arg method");
    }
    add(num1,num2){
        console.log("inside two arg method");
    }
}
var cal=new Calc()
cal.add(10,20)
cal.add(10)
//only work recently implemented method